### PR TITLE
[KYUUBI #6299] Support disabling Web UI

### LINKED ISSUE #6299: [TASK][EASY] Support disabling Web UI
### What's the level of this task?

EASY

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before creating

- [X] I have searched in the [task list](https://github.com/orgs/apache/projects/343) and found no similar tasks.

### Mentor

- [ ] I have sufficient expertise on this task, and I volunteer to be a mentor of this task to guide contributors through the task.

### Skill requirements

Scala programming language, Jetty.

### Background and Goals

Kyuubi has enabled RESTful protocol by default since 1.8, the UI is auto-enabled when RESTful protocol is enabled, for some reasons, user may want to disable UI while enabling RESTful protocol.

### Implementation steps

Make enabling Web UI configurable

### Additional context

Introduction of [2024H1 Kyuubi Code Contribution Program](https://github.com/apache/kyuubi/issues/6232)

### PR DESCRIPTION
**Pull request by lsm1**
# :mag: Description
## Issue References 🔗

This pull request fixes #6299 

## Describe Your Solution 🔧

when disabling web ui, return 404 page

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Related Unit Tests

---

# Checklist 📝

- [ ] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

**Be nice. Be informative.**

### CODE REVIEW

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 1]
```diff
@@ -626,6 +626,13 @@ object KyuubiConf {
       .timeConf
       .createWithDefaultString("PT5S")
 
+  val FRONTEND_REST_ENABLE_WEBUI: ConfigEntry[Boolean] =
+    buildConf("kyuubi.frontend.rest.enable.webui")
+      .doc("If set to false then web ui will disabled when RESTful protocol is enabled")
```
Comment (pan3793) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:631]:
```suggestion
    buildConf("kyuubi.frontend.rest.ui.enabled")
      .doc("Whether to enable Web UI when RESTful protocol is enabled")
```

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiRestFrontendService.scala [hunk 2]
```diff
@@ -116,6 +116,13 @@ class KyuubiRestFrontendService(override val serverable: Serverable)
   }
 
   private def installWebUI(): Unit = {
+    if (!conf.get(FRONTEND_REST_ENABLE_WEBUI)) {
```
Comment (pan3793) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiRestFrontendService.scala:119]: prefer to judge on the caller's side to make this function clear. BTW, what will happen if we don't inject the `enable.html` page?

Comment (lsm1) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiRestFrontendService.scala:119]: If we don't inject the `enable.html`, we just get a 404 page.

File: kyuubi-server/web-ui/public/enable.html [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <title>Apache Kyuubi</title>
+  </head>
+  <body>
+    <h2 style="text-align: center;">🙁 The Web UI is currently unavailable.</h2>
+    <h4 style="text-align: center;">👉 To enable Web UI, set kyuubi.frontend.rest.enable.webui true.</h4>
```
Comment (pan3793) [inline kyuubi-server/web-ui/public/enable.html:27]: the page may not much help as we did for `dist` missing `--web-ui`, let's keep 404 now

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merged to master

**Pull request closed without merging**